File: backend/app/schemas/simulator.py
"""Simulator request/response schemas."""

from pydantic import BaseModel


class IngredientAdjustment(BaseModel):
    ingredient_id: int
    new_quantity: float | None = None
    new_unit_price: float | None = None


class SimulateRequest(BaseModel):
    new_selling_price: float | None = None
    ingredient_adjustments: list[IngredientAdjustment] | None = None
    estimated_weekly_sales: int | None = None


class SimulatedIngredient(BaseModel):
    ingredient_id: int
    ingredient_name: str
    quantity: float
    unit: str
    unit_price: float | None
    line_cost: float | None
    changed: bool


class SimulationState(BaseModel):
    selling_price: float
    food_cost: float
    food_cost_percent: float
    margin_status: str  # green/orange/red
    gross_margin: float
    ingredients: list[SimulatedIngredient]


class SimulateResponse(BaseModel):
    recipe_name: str
    current: SimulationState
    simulated: SimulationState
    monthly_impact: float | None = None

File: frontend/src/types/simulator.ts
// Mirror of backend/app/schemas/simulator.py

export interface SimulatedIngredient {
  ingredient_id: number;
  ingredient_name: string;
  quantity: number;
  unit: string;
  unit_price: number | null;
  line_cost: number | null;
  changed: boolean;
}

export interface SimulationState {
  selling_price: number;
  food_cost: number;
  food_cost_percent: number;
  margin_status: string;
  gross_margin: number;
  ingredients: SimulatedIngredient[];
}

export interface SimulateResponse {
  recipe_name: string;
  current: SimulationState;
  simulated: SimulationState;
  monthly_impact: number | null;
}

export interface IngredientAdjustment {
  ingredient_id: number;
  new_quantity?: number | null;
  new_unit_price?: number | null;
}

export interface SimulateRequest {
  new_selling_price?: number | null;
  ingredient_adjustments?: IngredientAdjustment[] | null;
  estimated_weekly_sales?: number | null;
}
